### PR TITLE
docs: Add section for Secrets Manager Plugin

### PR DESCRIPTION
**Pull request by crystall-bitquill**
### Summary

Add a section for the Secrets Manager plugin in the documentation

### Description

Added a section for the Secrets Manager plugin in the README.md file including:
- What it is
- Required properties
- Sample code on how to use the plugin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -495,6 +495,54 @@ You can include additional monitoring configurations by adding the prefix `monit
...
+   private static final String SECRET_ID = "secretId";
+   private static final String REGION = "us-east-1";
+
+   public static void main(String[] args) throws SQLException {
+      final Properties properties = new Properties();
+      // Enable the AWS Secrets Manager Plugin
+      properties.setProperty(
+              "connectionPluginFactories",
+              String.format("%s",
+                      AWSSecretsManagerPluginFactory.class.getName()));
```
Comment (karenc-bq) [inline README.md:527]: We don't need to use string format if we are only using 1 connection plugin

File: README.md [hunk 2]
```diff
@@ -495,6 +495,54 @@ You can include additional monitoring configurations by adding the prefix `monit
...
+      
+      // Set these properties so secrets can be retrieved
+      properties.setProperty("secretsManagerSecretId", SECRET_ID);
+      properties.setProperty("secretsManagerRegion", REGION);
+
+      // Try and make a connection
+      try (final Connection conn = DriverManager.getConnection(CONNECTION_STRING, properties);
+              final Statement myQuery = conn.createStatement()) {
+         try (final ResultSet rs = myQuery.executeQuery("SELECT * FROM employees")) {
```
Comment (karenc-bq) [inline README.md:536]: These three can all be in the same try statement

**Review by sergiyv-improving — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -495,6 +495,50 @@ You can include additional monitoring configurations by adding the prefix `monit
...
+      // Enable the AWS Secrets Manager Plugin
+      properties.setProperty("connectionPluginFactories", AWSSecretsManagerPluginFactory.class.getName());
+      
+      // Set these properties so secrets can be retrieved
+      properties.setProperty("secretsManagerSecretId", SECRET_ID);
+      properties.setProperty("secretsManagerRegion", REGION);
+
+      // Try and make a connection
+      try (final Connection conn = DriverManager.getConnection(CONNECTION_STRING, properties);
+           final Statement myQuery = conn.createStatement();
```
Comment (karenc-bq) [inline README.md:532]: What do you think of renaming the variable to `statement`
```suggestion
           final Statement statement = conn.createStatement();
```

File: README.md [hunk 4]
```diff
@@ -495,6 +495,50 @@ You can include additional monitoring configurations by adding the prefix `monit
 > 
 > It is suggested to turn off Enhanced Failure Monitoring plugin, or to avoid using RDS Proxy endpoints when the plugin is active. 
 
+## AWS Secrets Manager Plugin
+
+The AWS JDBC Driver for MySQL supports usage of database credentials stored in the AWS Secrets Manager through the AWS Secrets Manager Plugin. This plugin is optional and can be enabled with the `connectionPluginFactories` parameter as seen in the [connection plugin manager parameters table](#connection-plugin-manager-parameters). When a user creates a new connection with this plugin enabled, the plugin will retrieve the secret and the connection will be created using those credentials.
```
Comment (karenc-bq) [inline README.md:500]:
```suggestion
The AWS JDBC Driver for MySQL supports usage of database credentials stored in the [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) through the AWS Secrets Manager Plugin. This plugin is optional and can be enabled with the `connectionPluginFactories` parameter as seen in the [connection plugin manager parameters table](#connection-plugin-manager-parameters). When a user creates a new connection with this plugin enabled, the plugin will retrieve the secret and the connection will be created using those credentials.
```

**Review by karenc-bq — APPROVED** [review]

**Pull request merged**